File: Lab 6/Lab 6.py
import os

def main():
    print_sum()

def print_sum():
    try:
        a = float(input("Gunawardana Shiron. IS-92. Please enter a, b, c.\na = "))
        b = float(input("b = "))
        c = float(input("c = "))
        T = ex_sum(a, b, c)
        print(T)
    except KeyboardInterrupt:
        print("\nProgram executed.")

def ex_sum(a, b, c):
    return (maxN(a, a + b) + maxN(a, b + c)) / (1 + maxN(a + b * c, 1.15))

File: Lab 6/Lab 6.cpp
#include <iostream>

using namespace std;

void Numerals(float*, float*, float*); // Ініціалізація змінних a, b, c.
float MaxN(float, float); // Знаходження максимального з чисел.
void Calculate(float, float, float); // Обчислення значення виразу.

int main() {
	float a, b, c; // Задаємо змінні.
	Numerals(&a, &b, &c);
	Calculate(a, b, c);
	return 0;
}

void Numerals(float* a, float* b, float* c) {
	cout << "Gunawardana Shiron. IS-92.\nPlease enter numbers a, b and c.\na = ";
	cin >> *a;
	cout << "b = ";
	cin >> *b;
	cout << "c = ";
	cin >> *c;
}

float MaxN(float a, float b) {
	return (a > b) ? a : b;
}

void Calculate(float a, float b, float c) {
	cout << (MaxN(a, a + b) + MaxN(a, b + c)) / (1 + MaxN(a + b * c, 1.15)) << endl;
}
